### ISSUE TITLE
XUnit.ITestOutputHelper.WriteLine not showing up

### ISSUE THREAD

**Issue by RMacfarlane** [formulahendry/vscode-dotnet-test-explorer#94]
_From @fractionalJoe on June 15, 2018 22:19_

Issue Type: <b>Bug</b>

When running XUnit test with ITestOutputHelper to write output, the output is not available in the .Net Test Log window. 
### Code
    public class FunctionHandlerTest
    {
        #region fields
        private readonly ITestOutputHelper Output;
        #endregion

        #region ctors
        public FunctionHandlerTest(ITestOutputHelper output)
        {
            Output = output;
        }
        #endregion

        #region methods
        [Fact]
        public void OutputTest()
        {
            Output.WriteLine(JsonConvert.SerializeObject(response, Formatting.Indented));
        }
        #endregion
    }
#### (Edited slightly to replace names of customer' sensitive data with "....")

> Running test .......
> 
> Microsoft (R) Build Engine version 15.7.179.6572 for .NET Core
> Copyright (C) Microsoft Corporation. All rights reserved.
> 
>   .....
> 
> Build succeeded.
>     0 Warning(s)
>     0 Error(s)
> 
> Time Elapsed 00:00:01.75
> 
> 
> [xUnit.net 00:00:00.3807450]   Discovering: .....
> [xUnit.net 00:00:00.4582069]   Discovered:  .....
> [xUnit.net 00:00:00.4131600]   Starting:    .....
> [xUnit.net 00:00:00.5760216]   Finished:    .....
> .....: passed
> 
> Total tests: 1. Passed: 1. Failed: 0. Skipped: 0

VS Code version: Code 1.24.1 (24f62626b222e9a8313213fb64b10d741a326288, 2018-06-13T17:47:35.732Z)
OS version: Linux x64 4.15.0-23-generic

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-4710MQ CPU @ 2.50GHz (8 x 2493)|
|GPU Status|2d_canvas: unavailable_software<br>flash_3d: unavailable_software<br>flash_stage3d: unavailable_software<br>flash_stage3d_baseline: unavailable_software<br>gpu_compositing: unavailable_software<br>multiple_raster_threads: enabled_on<br>native_gpu_memory_buffers: disabled_software<br>rasterization: unavailable_software<br>video_decode: unavailable_software<br>video_encode: unavailable_software<br>vpx_decode: unavailable_software<br>webgl: unavailable_off<br>webgl2: unavailable_off|
|Load (avg)|1, 1, 1|
|Memory (System)|15.57GB (7.71GB free)|
|Process Argv|/usr/share/code/code --unity-launch|
|Screen Reader|no|
|VM|0%|

</details><details><summary>Extensions (3)</summary>

Extension|Author (truncated)|Version
---|---|---
dotnet-test-explorer|for|0.3.1
docomment|k--|0.1.0
csharp|ms-|1.15.2

</details>

_Copied from original issue: Microsoft/vscode#52079_

**Comment by stefanforsberg** [comment]
Hmm, we just write what the dotnet test command outputs. At least that's what we're supposed to do.

If you execute dotnet test on the project outside of vscode does ITestOutputHelper output show up then?

**Comment by fractionalJoe** [comment]
No, it doesn't. It shows up in the test output in Visual Studio. This [page](https://xunit.github.io/docs/capturing-output) mentions that it's "wrapped up in the XML output". Unfortunately I don't know what XML output it's referring to.

**Comment by stefanforsberg** [comment]
Ah ok, at the moment we only print the output from dotnet test so I think this is more of an feature request than a bug.

**Comment by fractionalJoe** [comment]
makes sense. thanks for the help!

**Comment by wizofaus** [comment]
Where ARE we supposed to find the output in Visual Studio? The test results window shows nothing at all and the test explorer window shows no "Output" link for any test.

**Comment by andregomars** [comment]
Please, when we can have this feature available?

**Comment by sagron** [comment]
> No, it doesn't. It shows up in the test output in Visual Studio. This [page](https://xunit.github.io/docs/capturing-output) mentions that it's "wrapped up in the XML output". Unfortunately I don't know what XML output it's referring to.

I would like this too - the xml output is the trx file. Example of section below with my two lines of output (is inside TestRun root tag):
```xml
  <Results>
    <UnitTestResult executionId="374230f5-e734-4b8e-b35c-5281b317cb79" testId="f42eb502-f832-5d95-7bdf-686cf8c50376" testName="WorldTest.HexTests.PlanetTest.Draw" computerName="SAGRON-DEVEL" duration="00:00:04.1650000" startTime="2019-11-15T06:03:19.2771373-06:00" endTime="2019-11-15T06:03:19.2771417-06:00" testType="13cdc9d9-ddb5-4fa4-a97d-d965ccfc6d4b" outcome="Passed" testListId="8c84fa94-04c1-424b-9868-57a2d4851a1d" relativeResultsDirectory="374230f5-e734-4b8e-b35c-5281b317cb79">
      <Output>
        <StdOut>823436 points added in 2638.4ms.&#xD;
823436 lookups and pixel edits in 1474.9ms.</StdOut>
      </Output>
    </UnitTestResult>
  </Results>
```

**Comment by bartekmi** [comment]
FYI... The output only shows up if the test fails.

Please see my answer to this related question: https://stackoverflow.com/questions/64317627/no-output-in-vs-code-when-running-tests/71104818#71104818

**Comment by tarekahf** [comment]
I am facing the same issue. Any progress on this report? We need to debug the tests by writing to the log and be able to see the output in vs code.
